src/extension/php_objects.c

```diff
@@ -119,10 +157,79 @@ static zend_string *nullable _fetch_from_env(const char *name, size_t name_len)
     }
     *w = '\0';
 
+    return env_name;
+}
+
+/**
+ * This function is based on fpm_php_zend_ini_alter_master
+ * from php-src
+ */
+int _custom_php_zend_ini_alter_master(char *name, size_t name_length,
+    // NOLINTNEXTLINE(bugprone-easily-swappable-parameters)
+    char *new_value, size_t new_value_length, int mode, int stage,
+    bool has_env) /* {{{ */
+{
+    zend_ini_entry *ini_entry;
+    zend_string *duplicate;
+
+    if ((ini_entry = zend_hash_str_find_ptr(
+             EG(ini_directives), name, name_length)) == NULL) {
+        return FAILURE;
+    }
+
+    duplicate = zend_string_init(new_value, new_value_length, 1);
+
+    struct entry_ex *eex = ini_entry->mh_arg3;
+    eex->has_env = has_env;
+
+    if (!ini_entry->on_modify ||
+        ini_entry->on_modify(ini_entry, duplicate, ini_entry->mh_arg1,
+            ini_entry->mh_arg2, ini_entry->mh_arg3, stage) == SUCCESS) {
+        ini_entry->value = duplicate;
+        /* when mode == ZEND_INI_USER keep unchanged to allow ZEND_INI_PERDIR
+         * (.user.ini) */
+        if (mode == ZEND_INI_SYSTEM) {
+            ini_entry->modifiable = mode;
+        }
+    } else {
+        zend_string_free(duplicate);
+    }
+
+    return SUCCESS;
+}
+
+dd_result dd_phpobj_load_env_values()
+{
+    struct _dd_registered_entries current;
+    zend_string *env_def;
+    while (registered_entries_count > 0) {
+        current = registered_entries[--registered_entries_count];
+        env_def = _fetch_from_env(
+            registered_entries[registered_entries_count].env_name);
+        if (env_def) {
+            _custom_php_zend_ini_alter_master(current.name,
+                strlen(current.name), ZSTR_VAL(env_def), ZSTR_LEN(env_def),
+                PHP_INI_SYSTEM, PHP_INI_STAGE_RUNTIME, true);
```

Comment:
If you change current registered_entries.* to use `zend_string` you don't need to strlen, which adds up.

Reply:
Solved